### PR TITLE
Make the PM cron jobs clear the DAOs they've reduced

### PR DESCRIPTION
**Pull request by mcarcaso**
Also re-enable PMing but use a NullDAO for individual PMs.

### CODE REVIEW

File: src/cronjobs [hunk 1]
```diff
@@ -2,6 +2,15 @@ p({"class":"foam.nanos.cron.Cron","minute":1,"hour":-1,"dayOfMonth":-1,"month":-
...
-p({"class":"foam.nanos.cron.Cron","minute":0,"id":"PMHourlyReduce","enabled":false,"code":"rm = x.get(\"pmHourlyReduceManager\");\nrm.doReduce();"})
-p({"class":"foam.nanos.cron.Cron","hour":0,"id":"PMDailyReduce","enabled":false,"code":"rm = x.get(\"pmDailyReduceManager\");\nrm.doReduce();"})
-p({"class":"foam.nanos.cron.Cron","enabled":true,"id":"PmReduceManager","code":"import foam.nanos.logger.Logger;\n\nscriptName = \"PMReduceManager\";\nlogger = x.get(\"logger\");\ndailyRM = x.get(\"pmDailyReduceManager\");\nhourlyRM = x.get(\"pmHourlyReduceManager\");\nnow = java.time.LocalDateTime.now();\n\nif (now.getHour() == 0) {\n  hourlyRM.doReduce();\n  dailyRM.doReduce();\n  logger.info(new String[] {scriptName, \"Daily PM Reduce Complete\"});\n} else if (now.getMinute() == 0){\n  hourlyRM.doReduce();\n  logger.info(new String[] {scriptName, \"Hourly PM Reduce Complete\"});\n}\n"})
+p({"class":"foam.nanos.cron.Cron","minute":0,"id":"PMHourlyReduce","enabled":true,"code":"""
+rm = x.get(\"pmHourlyReduceManager\");
```
Comment (TW80000) [inline src/cronjobs]: I don't think strings need to be escaped in multi-line journal format.

File: src/cronjobs [hunk 2]
```diff
@@ -2,6 +2,15 @@ p({"class":"foam.nanos.cron.Cron","minute":1,"hour":-1,"dayOfMonth":-1,"month":-
...
-p({"class":"foam.nanos.cron.Cron","minute":0,"id":"PMHourlyReduce","enabled":false,"code":"rm = x.get(\"pmHourlyReduceManager\");\nrm.doReduce();"})
-p({"class":"foam.nanos.cron.Cron","hour":0,"id":"PMDailyReduce","enabled":false,"code":"rm = x.get(\"pmDailyReduceManager\");\nrm.doReduce();"})
-p({"class":"foam.nanos.cron.Cron","enabled":true,"id":"PmReduceManager","code":"import foam.nanos.logger.Logger;\n\nscriptName = \"PMReduceManager\";\nlogger = x.get(\"logger\");\ndailyRM = x.get(\"pmDailyReduceManager\");\nhourlyRM = x.get(\"pmHourlyReduceManager\");\nnow = java.time.LocalDateTime.now();\n\nif (now.getHour() == 0) {\n  hourlyRM.doReduce();\n  dailyRM.doReduce();\n  logger.info(new String[] {scriptName, \"Daily PM Reduce Complete\"});\n} else if (now.getMinute() == 0){\n  hourlyRM.doReduce();\n  logger.info(new String[] {scriptName, \"Hourly PM Reduce Complete\"});\n}\n"})
+p({"class":"foam.nanos.cron.Cron","minute":0,"id":"PMHourlyReduce","enabled":true,"code":"""
```
Comment (TW80000) [inline src/cronjobs]: It would be nice for readability to put all properties on their own line.

**Review by TW80000 — COMMENTED** [review]
This still doesn't appear to clear `pmDAO` at any point, which I think is the biggest offender.

File: src/cronjobs (hunk 1 above)
Comment (mcarcaso) [reply src/cronjobs]: Yup. I opened this PR a little too eagerly :p

File: src/cronjobs (hunk 2 above)
Comment (mcarcaso) [reply src/cronjobs]: Done

**Review by TW80000 — APPROVED** [review]
Haven't tested but the changes seem reasonable to me.

**Comment by mcarcaso** [comment]
> Haven't tested but the changes seem reasonable to me.

Yeah I tested locally but not over a long period of time. Time will tell :p

**Pull request merged**